src/pull.rs

```diff
@@ -194,7 +195,7 @@ impl<'a> PullClient<'a> {
 
         let destination = Path::new(&store_path);
 
-        if let Err(e) = unpack(out, destination) {
+        if let Err(e) = unpack(out.as_slice(), destination) {
             fs::remove_dir_all(destination)?;
```

Comment:
not related to this PR. It seems an error handling process. If `fs::remove_dir_all(destination)` fails, the reported error might not include enough error information for the context. I'm afraid we should add some context info like
```rust
fs::remove_dir_all(destination).context("Failed to roll back when unpacking")?
```

Reply:
Good suggestion, fixed now!